packages/keystone/fields/AutoIncrementInteger/Implementation.js

```diff
@@ -1,14 +1,23 @@
 const { Integer } = require('@keystonejs/fields')
+const get = require('lodash/get')
 const { default: Redlock } = require('redlock')
 
-
 const { getRedisClient } = require('@open-condo/keystone/redis')
 
 class AutoIncrementInteger extends Integer.implementation {
-    
+
     async resolveInput ({ resolvedData, existingItem }) {
         if (!existingItem && !resolvedData[this.path]) {
-            const next = await this.adapter.nextValue()
+            const scopeFields = get(this, ['config', 'autoIncrementScopeFields'], [])
+            const scopeWhere = scopeFields.reduce((result, field) => {
+                const fieldValue = get(resolvedData, field)
+                if (fieldValue) {
+                    return { ...result, [field]: fieldValue }
```

Comment:
Copy inside reduce? How about assigning property to result and just return result itself?

Reply:
I'd like the reducer function to be pure, to make the code functional-ish. Yes, there are no disadvantages if I use mutating of the result object here. The `scopeFields` object always will be small, so there are no pros/cons for these two approaches for this task.